### ISSUE TITLE
warn on `version: 3.10` (should be string)?

### ISSUE THREAD

**Issue by unhammer** [adrienverge/yamllint#738]
```
$ python3 -c 'import yaml; print(yaml.safe_load("version: 3.10"))'
{'version': 3.1}
# not good


$ git log -1 --pretty=%H
263fb8843771e560a818f7263928b775e73cd0c0


# but it seems like we get no warnings on it:
$ echo 'version: 3.10' | python3 yamllint -
stdin
  1:1       warning  missing document start "---"  (document-start)

```

**Comment by andrewimeson** [comment]
I don't think that yamllint is the right place to warn on this type of data issue.

Someone could legitimately want to display the level of precision of a measurement in their YAML document, even if the actual data when loaded doesn't keep that precision.

```yaml
---
engines:
  chevy:
    lq4:
      bore_diameter_inches: 4.000
      stroke_inches: 3.622
      deck_height_inches:
        lower_limit: 9.230
        upper_limit: 9.240
      goes_fast: true
```

```json
{
  "engines": {
    "chevy": {
      "lq4": {
        "bore_diameter_inches": 4,
        "stroke_inches": 3.622,
        "deck_height_inches": {
          "lower_limit": 9.23,
          "upper_limit": 9.24
        },
        "goes_fast": true
      }
    }
  }
}
```

This is better handled by governing the data schema (e.g. using JSON Schema) to enforce that `version` is a string instead of a floating point or integer.
